colossalai/inference/config.py

```diff
@@ -108,6 +109,9 @@ def _verify_config(self) -> None:
             self.dtype in _ALLOWED_DTYPES
         ), f"Expected dtype to be in {_ALLOWED_DTYPES} but found an unknown dtype: {self.dtype}"
 
+        if self.dtype == torch.float32:
+            self.high_precision = False
+
```

Comment:
Might want to refine the comment of `high_precision`. It's sort of ambiguous here that setting `high_precision` as `False` for data type float32. Maybe you want to express "skip using casting when the data type is float32"?

Reply:
Yes.

Reply:
Enabling high_precision for data type float32 is meaningless.